js/ui/grid_core/ui.grid_core.virtual_scrolling.js

```diff
@@ -161,14 +164,19 @@ const VirtualScrollingDataSourceAdapterExtender = (function() {
             }
         },
         _handleLoadError: function() {
-            const that = this;
+            this._isLoading = false;
+            this.loadingChanged.fire(false);
 
-            that._isLoading = false;
-            that.loadingChanged.fire(false);
-
-            that.callBase.apply(that, arguments);
+            this.callBase.apply(this, arguments);
         },
         _handleDataChanged: function(e) {
+            if(this.option(NEW_SCROLLING_MODE)) {
+                this._items = this._dataSource.items();
+                this.endPageIndex(this._dataSource.pageIndex());
```

Comment:
Why is _lastPageIndex assigned here?

Reply:
endPageIndex is used in the editing module. So, the endPageIndex should return dataSource.pageIndex() in a new scrolling mode.